.storybook/decorators/withLifecycle.ts

```diff
@@ -0,0 +1,54 @@
+const setDisabledState = (element: HTMLButtonElement, state: boolean) => {
+  element.disabled = state;
+  element.classList.toggle('ais-ClearRefinements-button--disabled');
+};
+
+export const withLifecycle = (
+  search: any,
+  container: HTMLElement,
+  fn: (node: HTMLElement) => void
+) => {
+  const actions = document.createElement('div');
+  actions.style.marginBottom = '15px';
+  actions.style.paddingBottom = '15px';
+  actions.style.borderBottom = '1px solid #e4e4e4';
+
+  const description = document.createElement('p');
+  description.textContent =
+    'Click on one of the button to add/remove the widget.';
+
+  const add = document.createElement('button');
+  add.style.marginRight = '10px';
+  add.className = 'ais-ClearRefinements-button';
+  add.textContent = 'Add widget';
+
+  const remove = document.createElement('button');
+  remove.className = 'ais-ClearRefinements-button';
+  remove.textContent = 'Remove widget';
+
+  const node = document.createElement('div');
+
+  actions.appendChild(description);
+  actions.appendChild(add);
+  actions.appendChild(remove);
+
+  container.appendChild(actions);
+  container.appendChild(node);
+
+  const widget = fn(node);
```

Comment:
weird that TS doesn't complain about this, since it should be void

Reply:
The widget has a type `void` which is completely valid. The `addWidget` & `removeWidget` are not typed which means that they are inferred as `any`. That's why it does not complain.